Task 3/Task3/Queue.cs

```diff
@@ -0,0 +1,61 @@
+﻿namespace Task3
+{
+    public class Queue<T> : IQueue<T>
+    {
+        private T[] _queue;
+        private int _head = 0;
+        private int _tail = 0;
+        private int _size;
+
+        public Queue(int size = 10)
+        {
+            _size = size;
+            _queue = new T[size];
+        }
+
+        public T Dequeue()
+        {
```

Comment:
Good solution idea, but not fully implemented. Think about if we make 10  Enqueue, then 10 Dequeue. Really stack is empty, but in you implementation _head = _tail = _size. and new Enque is not more possible. Please fix ))

Reply:
Fixed